Assets/Editor/ModelInspector.cs

```diff
@@ -511,29 +618,36 @@ public override void OnPreviewGUI(Rect r, GUIStyle background)
                 return;
             }
 
+            if (m_Settings.displayMode == DisplayMode.Shaded)
+            {
+                EditorGUI.DrawRect(r, new Color( 0.1f, 0.1f, 0.1f, 1));
+                DoLightSettings(r);
+                r.y += 20;
+            }
+
             Init();
 
             Assembly editorAssembly = Assembly.GetAssembly(typeof(EditorGUI));
             Type guiPreview = editorAssembly.GetType("PreviewGUI");
             MethodInfo drag2D = guiPreview.GetMethod("Drag2D");
 
-            if(m_Settings.displayMode != DisplayMode.UVLayout)
+            if (Event.current.button <= 0 && m_Settings.displayMode != DisplayMode.UVLayout)
                 m_Settings.previewDir = (Vector2)drag2D?.Invoke(null, new object[] {m_Settings.previewDir, r});
-            //previewDir = PreviewGUI.Drag2D(previewDir, r);
-            
-            if (Event.current.type == EventType.ScrollWheel && m_Settings.displayMode == DisplayMode.UVLayout)
+            //m_Settings.previewDir = PreviewGUI.Drag2D(m_Settings.previewDir, r);
+
+            if (Event.current.type == EventType.ScrollWheel)
                 MeshPreviewZoom(r, Event.current);
 
-            if (Event.current.type == EventType.MouseDrag && m_Settings.displayMode == DisplayMode.UVLayout)
+            if (Event.current.type == EventType.MouseDrag && (m_Settings.displayMode == DisplayMode.UVLayout || Event.current.button == 2))
```

Comment:
With all the panning & zooming, probably would be useful to add handling of `EventCommandNames.FrameSelected` command so that people could press F and get back to initial zoom/pan place.